### PR TITLE
Fixed target on link url in block editor - images

### LINKED ISSUE #25842: Block Editor: Open link in new window checkbox in the block editor. Don't save the selection.
### Parent Issue

_No response_

### Problem Statement

This selection can't be saved when checking or unchecking the option "Open link in new window" checkbox and adding an image into the block editor.

### Steps to Reproduce

1.- Add an image into the block editor.
2.- Add a URL link.
3.- Open link panel and uncheck the option "Open link in a new window."
4.- The uncheck selection still needs to be saved.

https://github.com/dotCMS/core/assets/12899960/b7083b0d-5829-4e6f-a971-ff809493716e

### Acceptance Criteria

The selection check/uncheck option should be saved.

### dotCMS Version

Demo
QA 23.01

### Proposed Objective

Technical User Experience

### Proposed Priority

Priority 2 - Important

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

https://dotcms.zendesk.com/agent/tickets/112923
https://dotcms.slack.com/archives/CSHTYUR7H/p1692642079655649

### Assumptions & Initiation Needs

_No response_

### Quality Assurance Notes & Workarounds

_No response_

### Sub-Tasks & Estimates

_No response_

### PR DESCRIPTION
**Pull request by KevinDavilaDotCMS**
### Summary
🖼️🛠️🎯

This pull request enhances the link functionality in the block editor library, especially for images. It fixes the default link target behavior, adds support for the `target` attribute in the `ImageNode` class, and improves the code quality of the `BubbleLinkFormView` class.

> _We are the masters of the image link_
> _We control the target of the click_
> _We defy the limits of the schema_
> _We unleash the power of the DOM_

### Walkthrough
*  Refactor `setLinkValues` method of `BubbleLinkFormView` class to improve readability and consistency ([link](https://github.com/dotCMS/core/pull/26321/files?diff=unified&w=0#diff-edefc37e869ad93708e241122fff269bbf2aa2921e9259efcbf3682bf575934bL188-R197))
*  Add `target` attribute to `ImageNode` schema, commands, and helper functions to enable opening image links in new tabs ([link](https://github.com/dotCMS/core/pull/26321/files?diff=unified&w=0#diff-82a4718dc9acbb94d2c87277edb662934d1578872b64ac21d92ddb51579068e0L13-R13), [link](https://github.com/dotCMS/core/pull/26321/files?diff=unified&w=0#diff-82a4718dc9acbb94d2c87277edb662934d1578872b64ac21d92ddb51579068e0R65-R69), [link](https://github.com/dotCMS/core/pull/26321/files?diff=unified&w=0#diff-0b838cef14c8ad6f040836023b4b0348fe64344220f19bebbce854dcffdd32fbL10-R12))
*  Add temporary console log statement and change default value of `blank` variable in `BubbleLinkFormView` class for debugging purposes ([link](https://github.com/dotCMS/core/pull/26321/files?diff=unified&w=0#diff-edefc37e869ad93708e241122fff269bbf2aa2921e9259efcbf3682bf575934bL245-R253))

### Additional Info
** any additional useful context or info **

### Screenshots
Original             |  Updated
:-------------------------:|:-------------------------:
** original screenshot **  |  ** updated screenshot **

### CODE REVIEW

File: core-web/libs/block-editor/src/lib/extensions/bubble-link-form/plugins/bubble-link-form.plugin.ts [hunk 1]
```diff
@@ -242,7 +246,10 @@ export class BubbleLinkFormView {
         const { href: link = '', target } = this.editor.isActive('link')
             ? this.editor.getAttributes('link')
             : this.editor.getAttributes(ImageNode.name);
-        const blank = target ? target === '_blank' : true;
+        const blank = target ? target === '_blank' : false;
+
+        // eslint-disable-next-line no-console
+        console.log(target);
```
Comment (zJaaal) [inline core-web/libs/block-editor/src/lib/extensions/bubble-link-form/plugins/bubble-link-form.plugin.ts:252]: do we need this?

File: core-web/libs/block-editor/src/lib/extensions/bubble-link-form/plugins/bubble-link-form.plugin.ts [hunk 2]
```diff
@@ -242,7 +246,10 @@ export class BubbleLinkFormView {
         const { href: link = '', target } = this.editor.isActive('link')
             ? this.editor.getAttributes('link')
             : this.editor.getAttributes(ImageNode.name);
-        const blank = target ? target === '_blank' : true;
+        const blank = target ? target === '_blank' : false;
```
Comment (zJaaal) [inline core-web/libs/block-editor/src/lib/extensions/bubble-link-form/plugins/bubble-link-form.plugin.ts:249]: `const blank = target === '_blank'` isn't this the same?

Comment (zJaaal) [reply core-web/libs/block-editor/src/lib/extensions/bubble-link-form/plugins/bubble-link-form.plugin.ts:249]: We can even default it to `false` or something else in the line 246 if it's possible

**Review by zJaaal — CHANGES_REQUESTED** [review]

File: core-web/libs/block-editor/src/lib/extensions/bubble-link-form/plugins/bubble-link-form.plugin.ts (hunk 1 above)
Comment (rjvelazco) [inline core-web/libs/block-editor/src/lib/extensions/bubble-link-form/plugins/bubble-link-form.plugin.ts:252]: Remove

File: core-web/libs/block-editor/src/lib/extensions/bubble-link-form/plugins/bubble-link-form.plugin.ts (hunk 2 above)
Comment (rjvelazco) [inline core-web/libs/block-editor/src/lib/extensions/bubble-link-form/plugins/bubble-link-form.plugin.ts:249]:
```suggestion
        const blank =  target === '_blank';
```

File: core-web/libs/block-editor/src/lib/nodes/image-node/image.node.ts [hunk 3]
```diff
@@ -10,7 +10,7 @@ declare module '@tiptap/core' {
             /**
              * Set Image Link mark
              */
-            setImageLink: (attributes: { href: string }) => ReturnType;
+            setImageLink: (attributes: { href: string; target: string | null }) => ReturnType;
```
Comment (rjvelazco) [inline core-web/libs/block-editor/src/lib/nodes/image-node/image.node.ts:13]:
```suggestion
            setImageLink: (attributes: { href: string; target?: string }) => ReturnType;
```

File: core-web/libs/block-editor/src/lib/extensions/bubble-link-form/plugins/bubble-link-form.plugin.ts [hunk 4]
```diff
@@ -242,7 +246,10 @@ export class BubbleLinkFormView {
         const { href: link = '', target } = this.editor.isActive('link')
```
Comment (rjvelazco) [inline core-web/libs/block-editor/src/lib/extensions/bubble-link-form/plugins/bubble-link-form.plugin.ts:246]:
```suggestion
        const { href: link = '', target = "_top" } = this.editor.isActive('link')
```

**Review by rjvelazco — CHANGES_REQUESTED** [review]

File: core-web/libs/block-editor/src/lib/extensions/bubble-link-form/plugins/bubble-link-form.plugin.ts (hunk 1 above)
Comment (fmontes) [reply core-web/libs/block-editor/src/lib/extensions/bubble-link-form/plugins/bubble-link-form.plugin.ts:252]: No :D

**Review by fmontes — CHANGES_REQUESTED** [review]

**Review by zJaaal — APPROVED** [review]

**Review by rjvelazco — APPROVED** [review]

**Pull request merged**